sklearn/utils/validation.py

```diff
@@ -239,19 +240,29 @@ def _ensure_sparse_format(spmatrix, accept_sparse, dtype, copy,
     spmatrix_converted : scipy sparse matrix.
         Matrix that is ensured to have an allowed type.
     """
-    if accept_sparse in [None, False]:
-        raise TypeError('A sparse matrix was passed, but dense '
-                        'data is required. Use X.toarray() to '
-                        'convert to a dense numpy array.')
     if dtype is None:
         dtype = spmatrix.dtype
 
     changed_format = False
-    if (isinstance(accept_sparse, (list, tuple))
-            and spmatrix.format not in accept_sparse):
-        # create new with correct sparse
-        spmatrix = spmatrix.asformat(accept_sparse[0])
-        changed_format = True
+
+    if isinstance(accept_sparse, str):
+        accept_sparse = [accept_sparse]
+
+    if accept_sparse in [None, False]:
+        raise TypeError('A sparse matrix was passed, but dense '
+                        'data is required. Use X.toarray() to '
+                        'convert to a dense numpy array.')
+    elif (isinstance(accept_sparse, (list, tuple)) and
+          len(accept_sparse)):
+        # ensure correct sparse format
+        if spmatrix.format not in accept_sparse:
+            # create new with correct sparse
+            spmatrix = spmatrix.asformat(accept_sparse[0])
+            changed_format = True
+    elif accept_sparse is not True:
+        # any other type
+        raise ValueError(("Invalid parameter "
+                          "'accept_sparse={}'").format(accept_sparse))
```

Comment:
Please list the allowed type of parameters, i.e. something like `'accept_sparse' should be a string, boolean or list of strings, you provided 'accept_sparse={}'.format(accept_sparse)`.